### PR TITLE
PR from office branch by Husan

### PR DESCRIPTION
**Pull request by KhusanKholmatov**
some changes in exceptions and updates according to comments on last office PR

### CODE REVIEW

File: src/main/java/com/exadel/coolDesking/exception/ErrorAdvice.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class ErrorAdvice {
+
+    @ExceptionHandler(NotFoundException1.class)
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/exception/ErrorAdvice.java]: Bad name for exception.

Comment (KhusanKholmatov) [reply src/main/java/com/exadel/coolDesking/exception/ErrorAdvice.java]: which one NotFoundException1, or ErrorAdvice?

File: src/main/java/com/exadel/coolDesking/exception/NotFoundException1.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.exadel.coolDesking.exception;
+
+import lombok.Getter;
+
+@Getter
+public class NotFoundException1 extends RuntimeException {
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/exception/NotFoundException1.java]: Main branch already has **NotFoundException**

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,7 +1,7 @@
-spring.datasource.url=jdbc:postgresql://localhost:5432/hotdesking
+spring.datasource.url=jdbc:postgresql://localhost:5432/HotDeskingSystem
 spring.datasource.driver-class-name=org.postgresql.Driver
```
Comment (viktarkolbik) [inline src/main/resources/application.properties:2]: don't change DB settings

File: src/main/java/com/exadel/coolDesking/office/OfficeController.java [hunk 4]
```diff
@@ -14,12 +17,21 @@
...
+        return ResponseEntity.status(HttpStatus.OK).body(new ArrayList(Arrays.asList(
+                officeResponse.getOffice(),
+                officeResponse.getMapIds()
+        )));
     }
 
-    @PostMapping
-    public ResponseEntity<Office> addOffice(@RequestBody @Valid Office office) {
-        return null;
+    @PostMapping("")
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeController.java:28]: Get rid of **("")**

File: src/main/java/com/exadel/coolDesking/exception/ConflictException.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.exadel.coolDesking.exception;
+
+import lombok.Getter;
+
+@Getter
+public class ConflictException extends RuntimeException {
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/exception/ConflictException.java:6]: Conflict Exception is already in the main branch

File: src/main/java/com/exadel/coolDesking/exception/NotFoundException.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package com.exadel.coolDesking.exception;
+
+import lombok.Getter;
+
+@Getter
+public class NotFoundException extends RuntimeException {
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/exception/NotFoundException.java:6]: Use the same from the main branch

File: src/main/java/com/exadel/coolDesking/office/OfficeController.java [hunk 7]
```diff
@@ -14,12 +17,21 @@
     private final OfficeService officeService;
 
     @GetMapping(path = "/{id}")
-    public ResponseEntity<Office> getById(@PathVariable UUID id) {
-        return null;
+    public ResponseEntity<?> getById(@PathVariable UUID id) {
+        OfficeResponse officeResponse = officeService.getOfficeById(id);
+        return ResponseEntity.status(HttpStatus.OK).body(new ArrayList(Arrays.asList(
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeController.java:22]: Why do you wrap list in another list?
Why do you put in the same ArrayList a list of offices and list of mapIds?

File: src/main/java/com/exadel/coolDesking/office/OfficeService.java [hunk 8]
```diff
@@ -1,21 +1,48 @@
...
+    public OfficeResponse create(OfficeCreateDto officeCreateDto) {
+        OfficeResponse officeResponse = new OfficeResponse();
+        if (officeRepository.existsByName(officeCreateDto.getName())){
+            throw new ConflictException("Office with this name already exists", Office.class, officeCreateDto.getName());
+        }
+        Office office = officeMapper.officeDtoToOffictEntity(officeCreateDto);
+        officeRepository.save(office);
+        officeResponse.setSuccess(true);
+        officeResponse.setMessage("OK");
+        return officeResponse;
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeService.java:32]: Your response must contain just created entity

File: src/main/java/com/exadel/coolDesking/office/OfficeController.java [hunk 9]
```diff
@@ -14,12 +17,21 @@
     private final OfficeService officeService;
 
     @GetMapping(path = "/{id}")
-    public ResponseEntity<Office> getById(@PathVariable UUID id) {
-        return null;
+    public ResponseEntity<?> getById(@PathVariable UUID id) {
+        OfficeResponse officeResponse = officeService.getOfficeById(id);
+        return ResponseEntity.status(HttpStatus.OK).body(Arrays.asList(
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeController.java:22]: Why not to return office response?

File: src/main/java/com/exadel/coolDesking/office/OfficeController.java [hunk 10]
```diff
@@ -14,12 +17,21 @@
...
+                officeResponse.getOffice(),
+                officeResponse.getMapIds()
+        ));
     }
 
     @PostMapping
-    public ResponseEntity<Office> addOffice(@RequestBody @Valid Office office) {
-        return null;
+    public ResponseEntity<?> addOffice(@RequestBody OfficeCreateDto officeCreateDto) {
+        return ResponseEntity.status(HttpStatus.CREATED).body(officeService.create(officeCreateDto).getMessage());
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeController.java]: return created office entry

File: src/main/java/com/exadel/coolDesking/office/OfficeService.java [hunk 11]
```diff
@@ -1,21 +1,43 @@
...
 
-    public Office create(Office office) {
-        return null;
+    public String create(OfficeCreateDto officeCreateDto) {
+        if (officeRepository.existsByName(officeCreateDto.getName())){
+            throw new ConflictException("Office with this name already exists", Office.class, officeCreateDto.getName());
+        }
+        Office office = officeMapper.officeDtoToOffictEntity(officeCreateDto);
+        officeRepository.save(office);
+        return "OK";
```
Comment (viktarkolbik) [inline src/main/java/com/exadel/coolDesking/office/OfficeService.java:29]: After you saved office entity, convert it to responseDto and return the Dto from this method

```suggestion
        return officeMapper.entityToDto(.....);
```

**Pull request closed without merging**